pkg/test/utils.go

```diff
@@ -242,3 +243,32 @@ func CompareGoldenObject(t *testing.T, p string, got []byte) {
 		t.Logf("wrote updated golden output to %s", p)
 	}
 }
+
+func PrettyPrintJSON[T any](t *testing.T, k T) string {
+	encoded, err := json.MarshalIndent(k, "", " ")
+	if err != nil {
+		t.Fatalf("error encoding to json: %v", err)
+	}
+
+	return string(encoded)
+}
+
+func PrettyPrintYAML[T any](t *testing.T, k T) string {
+	encoded, err := json.MarshalIndent(k, "", " ")
+	if err != nil {
+		t.Fatalf("error encoding to json: %v", err)
+	}
+
+	otherK := new(T)
+	err = json.Unmarshal(encoded, &otherK)
+	if err != nil {
+		t.Fatalf("error decoding from json: %v", err)
+	}
+
+	yEncoded, err := yaml.Marshal(otherK)
```

Comment:
Just curious: Why can't we just do the marshaling here?

Reply:
I think JSON serialization as an intermediate step is a good idea for normalization and type safety.